src/sentry/models/integration.py

```diff
@@ -31,23 +31,6 @@ class Meta:
         db_table = "sentry_pagerdutyservice"
 
 
-class PagerDutyServiceProject(Model):
-    __core__ = False
-
-    project = FlexibleForeignKey("sentry.Project", db_index=False, db_constraint=False)
-    pagerduty_service = FlexibleForeignKey("sentry.PagerDutyService")
-    organization_integration = FlexibleForeignKey("sentry.OrganizationIntegration", null=True)
```

Comment:
You need to convert these from fks -> ints first. That's why you're getting test failures here, since once we delete the model, django is no longer aware of it, but the db level constraints still exist and cause issues.
Take a look at the model changes I made here: https://github.com/getsentry/sentry/pull/16758/files
Specifically things like this: `incident = BoundedBigIntegerField(db_column="incident_id")`

What you'll need to do is do that for all of your foreign keys, then regenerate your migration. Your generated SQL should then just remove the constraints from the tables, and you can follow up with another PR that removes the model, then a third that removes the table.